### PR TITLE
Fixed Review Comments - Arrow Connector

### PR DESCRIPTION
**Pull request by lithinwxd**
## Contributor checklist

- [ ] Please make sure your submission complies with our [contributing guide](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md), in particular [code style](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#code-style) and [commit standards](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#commit-standards).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

General Changes
* ... :pr:`12345`
* ... :pr:`12345`

Hive Connector Changes
* ... :pr:`12345`
* ... :pr:`12345`
```

If release note is NOT required, use:

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/AbstractArrowFlightClientHandler.java [hunk 1]
```diff
@@ -43,14 +42,14 @@ public AbstractArrowFlightClientHandler(ArrowFlightConfig config)
         this.config = requireNonNull(config, "config is null");
     }
 
-    private ArrowFlightClient initializeClient(Optional<String> uri)
+    private ArrowFlightClient initializeClient(String uri)
```
Comment (elbinpallimalilibm) [inline presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/AbstractArrowFlightClientHandler.java:45]: I don't see a review comment to change this from `Optional` to plain string.

File: presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/AbstractArrowSplitManager.java [hunk 2]
```diff
@@ -46,7 +47,7 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHand
     {
         ArrowTableLayoutHandle tableLayoutHandle = (ArrowTableLayoutHandle) layout;
         ArrowTableHandle tableHandle = tableLayoutHandle.getTableHandle();
-        FlightDescriptor flightDescriptor = getFlightDescriptor(clientHandler.getConfig(),
+        FlightDescriptor flightDescriptor = getFlightDescriptor(arrowFlightConfig,
```
Comment (elbinpallimalilibm) [inline presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/AbstractArrowSplitManager.java:50]: no need to have the `ArrowFlightConfig` parameter for `getFlightDescriptor`

File: presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestArrowFlightQueriesWithDictionaryVector.java [hunk 3]
```diff
@@ -84,7 +84,7 @@ public void testDictionaryBlock()
         // Validate the row count first
         assertEquals(actual.getRowCount(), 3);
         // Now, validate each row
-        MaterializedResult expectedRow = resultBuilder(getSession(), createVarcharType(10), BigintType.BIGINT)
+        MaterializedResult expectedRow = resultBuilder(getSession(), VarcharType.VARCHAR, BigintType.BIGINT)
```
Comment (elbinpallimalilibm) [inline presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestArrowFlightQueriesWithDictionaryVector.java:87]: why would `createVarcharType(10)` not work?

File: presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestingArrowMetadata.java [hunk 4]
```diff
@@ -88,7 +88,7 @@ public List<SchemaTableName> listTables(ConnectorSession session, Optional<Strin
 
     public List<String> extractSchemaAndTableData(Optional<String> schema, ConnectorSession connectorSession)
     {
-        try (ArrowFlightClient client = clientHandler.getClient(Optional.empty())) {
+        try (ArrowFlightClient client = clientHandler.createArrowFlightClient(null)) {
```
Comment (elbinpallimalilibm) [inline presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestingArrowMetadata.java:91]: call the parameter less method

File: presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowPageSource.java [hunk 5]
```diff
@@ -63,7 +63,7 @@ private void getFlightStream(AbstractArrowFlightClientHandler clientHandler, byt
         try {
             Optional<String> uri = (split.getLocationUrls().isEmpty()) ?
                     Optional.empty() : Optional.of(split.getLocationUrls().get(0));
-            flightClient = clientHandler.getClient(uri);
+            flightClient = uri.isPresent() ? clientHandler.createArrowFlightClient(uri.get()) : clientHandler.createArrowFlightClient(null);
```
[the remark this answers is not in the document]
Comment (elbinpallimalilibm) [reply presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowPageSource.java:66]: can we not use ternary operator? an if..else would be readable

File: presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/AbstractArrowFlightClientHandler.java [hunk 6]
```diff
@@ -91,19 +90,19 @@ private synchronized void initializeAllocator()
...
+    public ArrowFlightClient createArrowFlightClient()
     {
-        return initializeClient(uri);
+        return initializeClient(null);
     }
 
     public FlightInfo getFlightInfo(FlightDescriptor flightDescriptor, ConnectorSession connectorSession)
     {
-        try (ArrowFlightClient client = getClient(Optional.empty())) {
+        try (ArrowFlightClient client = createArrowFlightClient(null)) {
```
[the remark this answers is not in the document]
Comment (lithinwxd) [reply presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/AbstractArrowFlightClientHandler.java]: done

File: presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/AbstractArrowSplitManager.java (hunk 2 above)
Comment (lithinwxd) [reply presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/AbstractArrowSplitManager.java]: changed

File: presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowPageSource.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (lithinwxd) [reply presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowPageSource.java]: done

File: presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestArrowFlightQueriesWithDictionaryVector.java (hunk 3 above)
Comment (lithinwxd) [reply presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestArrowFlightQueriesWithDictionaryVector.java:87]: The Field which we created doesn't have precision , the default data from tpch data had varchar(10) but now we have returned only varchar

File: presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestingArrowMetadata.java (hunk 4 above)
Comment (lithinwxd) [reply presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestingArrowMetadata.java]: done

File: presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestArrowFlightQueriesWithDictionaryVector.java (hunk 3 above)
Comment (elbinpallimalilibm) [reply presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestArrowFlightQueriesWithDictionaryVector.java:87]: can you import static `VarcharType.VARCHAR`

File: presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowSplit.java [hunk 7]
```diff
@@ -77,14 +74,8 @@ public String getTableName()
     }
 
     @JsonProperty
-    public byte[] getTicket()
+    public byte[] getByteArray()
```
Comment (elbinpallimalilibm) [inline presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowSplit.java:77]: rename this to flightendpoint

File: presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowSplit.java [hunk 8]
```diff
@@ -30,20 +30,17 @@ public class ArrowSplit
...
-    private final List<String> locationUrls;
+    private final byte[] byteArray;
 
     @JsonCreator
     public ArrowSplit(
             @JsonProperty("schemaName") @Nullable String schemaName,
             @JsonProperty("tableName") String tableName,
-            @JsonProperty("ticket") byte[] ticket,
-            @JsonProperty("locationUrls") List<String> locationUrls)
+            @JsonProperty("byteArray") byte[] byteArray)
```
Comment (elbinpallimalilibm) [inline presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowSplit.java:39]: rename this to flightendpoint

File: presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowSplit.java [hunk 9]
```diff
@@ -30,20 +30,17 @@ public class ArrowSplit
 {
     private final String schemaName;
     private final String tableName;
-    private final byte[] ticket;
-    private final List<String> locationUrls;
+    private final byte[] byteArray;
```
Comment (elbinpallimalilibm) [inline presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowSplit.java:33]: rename this to flightendpoint

File: presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestArrowSplit.java [hunk 10]
```diff
@@ -31,29 +38,35 @@ public class TestArrowSplit
...
     @Test
     public void testConstructorAndGetters()
+            throws IOException, URISyntaxException
     {
         // Test that the constructor correctly initializes fields
         assertEquals(arrowSplit.getSchemaName(), schemaName, "Schema name should match.");
         assertEquals(arrowSplit.getTableName(), tableName, "Table name should match.");
-        assertEquals(arrowSplit.getTicket(), ticket, "Ticket byte array should match.");
-        assertEquals(arrowSplit.getLocationUrls(), locationUrls, "Location URLs list should match.");
+        assertNotNull(arrowSplit.getByteArray());
```
Comment (elbinpallimalilibm) [inline presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestArrowSplit.java:69]: can you deserialize flightendpoint and assert the contents

**Pull request merged**